Troubleshooting.md

```diff
@@ -688,6 +690,21 @@ Following the completion of the aforementioned commands, you can proceed with th
 
 [Reference](https://stackoverflow.com/questions/72299444/docker-desktop-doesnt-install-saying-docker-ce-cli-not-installable) for the solution stated.
 
+### make commands: `Operation not permitted`
+
+Some users might face the following error while running `make` commands:
+```
+Operation not permitted
+make: *** [Makefile:14: docker] Error 1
+```
+
+The simple solution for the above error is to run the make commands with admin access i.e. with `sudo`, example:
+```
+sudo make build
+```
+The cause of the issue is that Docker daemon binds to a unix socker by default, which is owned by root. So, when you run the make commands without admin access, it is not able to access the unix socket and hence the error is thrown.
```

Comment:
The first "socker" is misspelled

Also, in what circumstances does this happen? Does it happen only for particular OS's? What differentiates machines on which `sudo` is necessary and on which it is not? (See comment above too, re trying to avoid this altogether.)

Reply:
it happens when the current user donot have root priviledge for accessing the UNIX socket. instead of adding `sudo` since it add evevated permissions, we can simply add a new docker user group. this fixes the issue for linux and macOS users. not is if this happens for windows also as no one has encountered such issue